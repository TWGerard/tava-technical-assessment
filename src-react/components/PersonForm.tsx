import { useForm } from "react-hook-form";

const PersonForm = ({
  person
}: {
  person: any
}) => {
  const { register, handleSubmit } = useForm({ defaultValues: person });

  return (
    <form onSubmit={handleSubmit(data => console.log(data))}>
      <div>
        <input {...register("firstName")} placeholder="First Name" />
        <input {...register("lastName")} placeholder="Last Name" />
        <input {...register("email")} placeholder="Email" />
        <input {...register("phone")} placeholder="Phone" maxLength={11} />
      </div>
      <div>
        <h3>Employee Details</h3>
        <input {...register("Employee.directoryId")} />
        <input {...register("Employee.title")} placeholder="Title" />
      </div>
      <div>
        <h3>User Details</h3>
        <select {...register("User.userType")}>
          <option value="user" selected={person.User?.userType == "user"}>User</option>
          <option value="employee" selected={person.User?.userType == "employee"}>Employee</option>
          <option value="admin" selected={person.User?.userType == "admin"}>Admin</option>
        </select>
      </div>

      <input type="submit" value={person.id ? "Save" : "Create"} />
    </form>
  );
};

export default PersonForm;
