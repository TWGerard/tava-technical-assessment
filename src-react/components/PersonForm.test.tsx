import { render } from '@testing-library/react';
import { MockedProvider } from "@apollo/client/testing/react";

import PersonForm from './PersonForm';

describe('PersonForm', () => {
  it('renders', async () => {
    const person = {};
    const onSubmit = () => { };

    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <PersonForm person={person} onSubmit={onSubmit} />
      </MockedProvider>
    );

    expect(asFragment).toMatchSnapshot();
  });
});
