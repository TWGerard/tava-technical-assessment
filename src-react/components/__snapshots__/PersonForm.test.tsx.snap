// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PersonForm renders 1`] = `[Function]`;
